### PR TITLE
CkAbort: allow format strings in CkArray, CkMigratable

### PR DESCRIPTION
**Pull request by matthiasdiener**
also clarifies manual

### CODE REVIEW

File: src/ck-core/cklocation.C [hunk 1]
```diff
@@ -1552,8 +1553,13 @@ CkMigratable::~CkMigratable() {
...
 
-void CkMigratable::CkAbort(const char *why) const {
-	::CkAbort("CkMigratable '%s' aborting: %s",_chareTable[thisChareType]->name, why);
+void CkMigratable::CkAbort(const char *format, ...) const {
+	char newmsg[256];
+	va_list args;
+	va_start(args, format);
+	vsnprintf(newmsg, sizeof(newmsg), format, args);
+	va_end(args);
+	::CkAbort("CkMigratable '%s' aborting: %s",_chareTable[thisChareType]->name, newmsg);
```
Comment (evan-charmworks) [inline src/ck-core/cklocation.C]: Would it be preferable to make the `CkMigratable` name message a separate call to `CkError` as in `ArrayElement`?

Comment (matthiasdiener) [reply src/ck-core/cklocation.C]: Yeah, that's a good idea. Changed in the most recent commit.

**Comment by matthiasdiener** [comment]
> The class-level `CkAbort` functions aren't actually passing the arguments other than the format string to their callees. They would need to use a construct like this:

Couldn't we just interpolate the string directly in the CkAbort class functions?

**Comment by evan-charmworks** [comment]
> > The class-level `CkAbort` functions aren't actually passing the arguments other than the format string to their callees. They would need to use a construct like this:
> 
> Couldn't we just interpolate the string directly in the CkAbort class functions (similar to what was done before the most recent commit for `CkMigratable::CkAbort`)?

Yes, that would also resolve the issue.

**Comment by jcphill** [comment]
This is dangerous because you are changing the interpretation of the string passed to CkAbort, which now needs to be scrubbed of spurious % characters. Can you change this so that if there is only one argument it is not treated as a format string, e.g. make CkAbort(msg) equivalent to CkAbort("%s",msg).

**Comment by jcphill** [comment]
If that isn't possible, the new behavior should be in a new function called CkAbortf.

**Pull request merged**